src/book/book.controller.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Patch,
+  Post,
+  Query,
+  UseGuards,
+} from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { Book } from './book.entity';
+import { BookService } from './book.service';
+import { BookDto } from './dto/book-dto';
+
+@Controller('book')
+@UseGuards(AuthGuard())
+export class BookController {
+  constructor(private bookService: BookService) {}
+  @Get()
+  getBooks(@Query('search') searchBook: string): Promise<Book[]> {
+    return this.bookService.getBooks(searchBook);
```

Comment:
Use @Query as a DTO because we will have may query parameters in the future.

Reply:
done